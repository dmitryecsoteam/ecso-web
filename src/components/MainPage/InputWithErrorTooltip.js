import React from 'react';
import classNames from 'classnames';

export default ({
    children,
    error = false,
    errorText = '',
    label = '',
    disabled = false
} = props) => {

    const divClassName = classNames(
        'search-form__label-input',
        { 'search-form__label-input--error': error },
        { 'search-form__label-input--disabled': disabled}
    );

    const errorClassName = classNames(
        'search-form__error',
        { 'search-form__error--open': error }
    );

    return (
        <div className="search-form__input-container">
            <div className={divClassName}>
                <label className="search-form__label">{label}</label>
                {children}
            </div>
            <div className={errorClassName}>
                <span className="search-form__error-text">{errorText}</span>
            </div>
        </div>
    );
}