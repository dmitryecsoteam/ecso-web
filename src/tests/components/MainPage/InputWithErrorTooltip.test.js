import React from 'react';
import { shallow } from 'enzyme';
import InputWithErrorTooltip from '../../../components/MainPage/InputWithErrorTooltip';

test('should render InputWithErrorTooltip component with input and default props', () => {
    const wrapper = shallow(<InputWithErrorTooltip>
        <input type="text" />
    </InputWithErrorTooltip>);
    expect(wrapper).toMatchSnapshot();
});

test('should render InputWithErrorTooltip component with input', () => {
    const wrapper = shallow(<InputWithErrorTooltip
        error={false}
        errorText={null}
        label="Test"
        disabled={false}
    >
        <input type="text" />
    </InputWithErrorTooltip>);
    expect(wrapper).toMatchSnapshot();
});

test('should render InputWithErrorTooltip disabled', () => {
    const wrapper = shallow(<InputWithErrorTooltip
        error={false}
        errorText={null}
        label="Test"
        disabled={true}
    >
        <input type="text" />
    </InputWithErrorTooltip>);
    expect(wrapper).toMatchSnapshot();
});

test('should render InputWithErrorTooltip with error message', () => {
    const wrapper = shallow(<InputWithErrorTooltip
        error={true}
        errorText="Please type text"
        label="Test"
        disabled={true}
    >
        <input type="text" />
    </InputWithErrorTooltip>);
    expect(wrapper).toMatchSnapshot();
});

test('should render InputWithErrorTooltip with error message if it\'s empty string', () => {
    const wrapper = shallow(<InputWithErrorTooltip
        error={true}
        errorText=""
        label="Test"
        disabled={true}
    >
        <input type="text" />
    </InputWithErrorTooltip>);
    expect(wrapper).toMatchSnapshot();
});

test('should render InputWithErrorTooltip with hint', () => {
    const wrapper = shallow(<InputWithErrorTooltip
        error={true}
        errorText=""
        label="Test"
        disabled={true}
        hint={true}
        hintText="Test hint text"
    >
        <input type="text" />
    </InputWithErrorTooltip>);
    expect(wrapper).toMatchSnapshot();
});