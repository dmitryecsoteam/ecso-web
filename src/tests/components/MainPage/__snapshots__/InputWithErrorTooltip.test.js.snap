// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render InputWithErrorTooltip component with input 1`] = `
<div
  className="search-form__input-container"
>
  <div
    className="search-form__label-input"
  >
    <label
      className="search-form__label"
    >
      Test
    </label>
    <input
      type="text"
    />
  </div>
</div>
`;

exports[`should render InputWithErrorTooltip component with input and default props 1`] = `
<div
  className="search-form__input-container"
>
  <div
    className="search-form__label-input"
  >
    <input
      type="text"
    />
  </div>
</div>
`;

exports[`should render InputWithErrorTooltip disabled 1`] = `
<div
  className="search-form__input-container"
>
  <div
    className="search-form__label-input search-form__label-input--disabled"
  >
    <label
      className="search-form__label"
    >
      Test
    </label>
    <input
      type="text"
    />
  </div>
</div>
`;

exports[`should render InputWithErrorTooltip with error message 1`] = `
<div
  className="search-form__input-container"
>
  <div
    className="search-form__label-input search-form__label-input--error search-form__label-input--disabled"
  >
    <label
      className="search-form__label"
    >
      Test
    </label>
    <input
      type="text"
    />
  </div>
  <div
    className="search-form__error search-form__error--open"
  >
    <span
      className="search-form__error-text"
    >
      Please type text
    </span>
  </div>
</div>
`;
